### PR TITLE
Milestone 3: Added React redux

### PR DESCRIPTION
**Pull request by prg-04**
### I have done the following to reach this milestone:

- [ ] Wrapped my `<App>` with the `<Provider>` component from react-redux
- [ ] Passed my configured store into the `<Provider>`
- [ ] Created an initialState variable for my booksSlice.
- [ ] Displayed my books, received from the slice, in a reusable component
- [ ] Dispatched actions using useDispatch
- [ ] Added a `<Button>` component, which includes:
  - [ ] An event handler that adds a book to the books array (with attributes id, title and author)
- [ ] Add a `<Button>` component, which includes:
  - [ ] An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/Home.jsx [hunk 1]
```diff
@@ -17,27 +23,36 @@ function Home() {
...
       </ul>
       <Form setBooks={setBooks} />
-    </>
+
+      {/* <Button
+        type="button"
+        content="test run button"
+        padding="0.5rem 1rem"
+        transform="uppercase"
+      /> */}
```
Comment (wayungi) [inline src/components/Home.jsx]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history. Kindly delete all instances of unused code left as comments 🤝

File: src/redux/book/bookSlice.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    removeBook: (state, action) => {
+      const index = state.findIndex((book) => book.id === action.payload.id);
+
+      if (index !== -1) {
+        state.splice(index, 1);
+      }
+    },
+  },
+});
```
Comment (wayungi) [inline src/redux/book/bookSlice.js:22]: - [OPTIONAL] While immer library is used behind the scene to  apply the changes in an immutable manner, for the sake of practice, I suggest you write immutable logic in the reducer. 😄

File: src/components/Form.jsx
Comment (wayungi) [inline src/components/Form.jsx:1]: - [ ] Please  make sure that the Form component has local state for title and author, which is updated through the onChange event. Please delete line 14..16 in favor of the approach below. The code snippets below should guide you. 👍🏾 

```
function Form() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")

```

- [ ] Please add onchange event listeners on the input fields on line 38 & 39 😸 
```
 <input className="form__input" type="text" placeholder="Book title" onChange= {(e) =setTitle(e.target.value)} />
 <input className="form__input" type="text" placeholder="Author"  onChange= {(e) =setTitle(e.target.value)}/>
 ```
 
 - [ ] Please also include an onClick event handler on your button to add the book to the books array. Note that this is a project requirement to implement it using this approach 🦅 
 
 ```
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
```

File: src/components/Book.jsx [hunk 3]
```diff
@@ -2,43 +2,57 @@ import React from 'react';
...
+                borderRight: btn !== 'Edit' ? '1px solid #aaa' : 'none',
+              }}
+              borderTRradius="0px"
+              borderBRradius="0px"
+              className="book_btn"
+              type="button"
+              content={btn}
+              padding="0.501rem 1.188rem 0.586rem 1.375rem"
+            />
+          ))}
```
Comment (wayungi) [inline src/components/Book.jsx]: - [ ] Good add adding all the buttons required. Please note that the **remove button** an event handler to **remove** a book from the `books` array (with the *id* attribute). 

```
<Button onClick={() => dispatch(removeBook(id))}>Remove</Button>`
```

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @prg-04 , 

image


**Congratulations on the good job done so far!** 🤝 

There are some issues that you still need to work on to go to the next project but you are almost there! 😄 

## Highlights ⚡ ⚡ 
- [ ] All linter checks passing ✔️ 
- [ ] Good commit messages ✔
- [ ]  Correct Git flow ✔️ 
- [ ] Descriptive PR body ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Please ensure that you can add and remove books in the application 😄 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.jsx [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 import PropTypes from 'prop-types';
 import SelectCategory from './SelectCategory';
+import { addBook } from '../redux/book/bookSlice';
 import { categories } from '../constants/constants';
 
-function Form({ setBooks }) {
-  const [selected, setSelected] = useState(categories[0]);
+function Form({ onAddBook }) {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
```
Comment (VICTOR-CHUKWU) [inline src/components/Form.jsx:11]: - [ ] [OPTIONAL] you can actually use one state to manage all your input values like 
```
 const [input, setInput] = useState({
author: '',
title: ''
});
```

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @prg-04 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] redux used correctly in components

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**